pdr_backend/pdr_dashboard/util/db.py

```diff
@@ -317,3 +368,153 @@ def predictoors_stats(self):
+                feed["contract"], self.feeds_payout_stats
+            )
+
+            feed_item.update(result)
+
+            subscription_result = get_feeds_subscription_stat_with_contract(
+                feed["contract"], self.feeds_subscriptions
+            )
+
+            feed_item.update(subscription_result)
+
+            new_feed_data.append(feed_item)
+
+        columns = get_feed_column_ids(new_feed_data[0])
+
+        formatted_data = format_table(new_feed_data, columns)
+
+        return columns, formatted_data, new_feed_data
+
+    def _format_for_predictoors_table(
+        self,
+    ) -> Tuple[List[Dict[str, str]], List[Dict[str, Any]], List[Dict[str, Any]]]:
+
+        temp_data = self.predictoors_data
+
+        new_predictoor_data = []
+
+        # split the pair column into two columns
+        for data_item in temp_data:
+            tx_costs = self.fee_cost * float(data_item["stake_count"])
+
+            temp_pred_item = {}
+            temp_pred_item["addr"] = str(data_item["user"])
+            temp_pred_item["full_addr"] = str(data_item["user"])
+            temp_pred_item["apr"] = data_item["apr"]
+            temp_pred_item["accuracy"] = data_item["avg_accuracy"]
+            temp_pred_item["number_of_feeds"] = str(data_item["feed_count"])
+            temp_pred_item["staked_(OCEAN)"] = data_item["total_stake"]
+            temp_pred_item["gross_income_(OCEAN)"] = data_item["gross_income"]
+            temp_pred_item["stake_loss_(OCEAN)"] = data_item["stake_loss"]
+            temp_pred_item["tx_costs_(OCEAN)"] = tx_costs
+            temp_pred_item["net_income_(OCEAN)"] = data_item["total_profit"] - tx_costs
+
+            new_predictoor_data.append(temp_pred_item)
+
+        columns = get_feed_column_ids(new_predictoor_data[0])
+
+        formatted_data = format_table(new_predictoor_data, columns)
+
+        return columns, formatted_data, new_predictoor_data
+
+    def filter_for_feeds_table(
+        self, predictoor_feeds_only, predictoors_addrs, search_value, selected_feeds
+    ):
+        filtered_data = self.feeds_data
+
+        # filter feeds by payouts from selected predictoors
+        if predictoor_feeds_only and (len(predictoors_addrs) > 0):
+            feed_ids = self.feed_ids_based_on_predictoors(
+                predictoors_addrs,
+            )
+            filtered_data = [
+                obj
+                for obj in filtered_data
+                if obj["contract"] in feed_ids
+                if obj not in selected_feeds
+            ]
+
+        # filter feeds by pair address
+        filtered_data = (
+            filter_objects_by_field(
+                self.feeds_data, "pair", search_value, selected_feeds
+            )
+            if search_value
+            else filtered_data
+        )
+
+        return selected_feeds + filtered_data
+
+    @enforce_types
+    def format_predictoors_home_page_table_data(self) -> List[Dict[str, Any]]:
```

Comment:
I suggest renaming this to "get_formatted_predictoors_home_page_teble_data" because as it is now it's not clear that it returns something, it suggest that it's only formatting and changing something inside dbGetter

Reply:
Good point! How about a compromise solution: name it "formatted_pre..." and marking it as a property? I think that's the best representation of what it does.

Reply:
that could also work